src/client/particles.h

```diff
@@ -196,26 +195,33 @@ friend class ParticleSpawner;
 	void addNodeParticle(IGameDef *gamedef, LocalPlayer *player, v3s16 pos,
 		const MapNode &n, const ContentFeatures &f);
 
-	u32 getSpawnerId() const
+	/**
+	 * This function is only used by client particle spawners
+	 *
+	 * We don't need to check the particle spawner list because client ID will n
+	 * ever overlap (u64)
+	 * @return new id
+	 */
+	u64 generateSpawnerId()
 	{
-		for (u32 id = 0;; ++id) { // look for unused particlespawner id
-	 		if (m_particle_spawners.find(id) == m_particle_spawners.end())
-	 			return id;
-	 	}
+		return m_next_particle_spawner_id++;
```

Comment:
i think it should actually suffice to have ids be `u32` (as before) and have client ids start at 0x7fffffff

Reply:
if i do this:

* server must never send a higher value => i should modify the server behaviour
* protocol must restrict the value to.

Are you sure you want that ?

Reply:
the protocol already uses `u64`?

Reply:
the protocol uses u32. I used u64 to keep the protocol and server code as-is.
If you prefer i modify the server part to split the id part in 2 areas and add the client securities i can, but honestly 1 u32 vs 1 u64 memory difference is very low. We don't have billions of particle spawners

Reply:
I misunderstood this change a little, should be fine as is.

Reply:
is this an approval ?

Reply:
not yet, i'll test it later